Validly.Tests/Validators/Strings/PasswordTests.cs

```diff
@@ -0,0 +1,174 @@
+using Validly.Extensions.Validators.Strings;
+
+namespace Validly.Tests.Validators.Strings;
+
+[Validatable]
+partial record PasswordNoRequirementsTestObject
+{
+	[Password(8, requireSpecialChar: false, requireUpperCase: false, requireDigit: false)]
+	public required string Password { get; init; }
+}
+
+[Validatable]
+partial record PasswordRequireSpecialCharTestObject
+{
+	[Password(8, requireSpecialChar: true, requireUpperCase: false, requireDigit: false)]
+	public required string Password { get; init; }
+}
+
+[Validatable]
+partial record PasswordRequireSpecialAndUpperTestObject
+{
+	[Password(8, requireSpecialChar: true, requireUpperCase: true, requireDigit: false)]
+	public required string Password { get; init; }
+}
+
+[Validatable]
+partial record PasswordRequireAllTestObject
+{
+	[Password(8, requireSpecialChar: true, requireUpperCase: true, requireDigit: true)]
+	public required string Password { get; init; }
+}
+
+
+public class PasswordTests
+{
+	[Theory]
+	[InlineData("password")]
+	[InlineData("PASSWORD1")]
+	[InlineData("12345678")]
+	[InlineData("!@#$%^&*")]
+	[InlineData("Mixed!Pass")]
+	public void ValidNoRequirementsPasswordTest(string password)
+	{
+		if (password.Length < 8) return;
+
+		var value = new PasswordNoRequirementsTestObject { Password = password };
+
+		using var result = value.Validate();
+
+		Assert.True(result.IsSuccess);
+	}
+
+	[Theory]
+	[InlineData("short")]
+	[InlineData("hi")]
+	[InlineData("")]
+	[InlineData(null)]
+	public void InvalidNoRequirementsPasswordTest(string? password)
+	{
+		var value = password is null
```

Comment:
Why is the ternary operator here? `Password = password!`
Or did you want to create different objects in case it's null? In both branches there is just `PasswordNoRequirementsTestObject`